pkg/operator/workloads/logs.go

```diff
@@ -40,423 +37,187 @@ const (
+			if writePending {
+				writeString(socket, "\npending...")
+				writePending = false
 			}
 
-			for logProcess := range initProcesses {
-				process, err := createKubectlProcess(StringToLogKey(logProcess), &procAttr)
+			if len(prefix) == 0 {
+				prefix, err = getPrefix(podLabels)
 				if err != nil {
-					socketWriterError <- err
-					return
+					writeString(socket, err.Error())
+					closeSocket(socket)
+					continue
 				}
-				processMap[logProcess] = process
 			}
 
-			if len(tfServingProcesses) != 0 && len(initProcesses) != 0 {
-				time.Sleep(100 * time.Millisecond)
+			if len(prefix) == 0 {
+				timer.Reset(pollPeriod)
+				continue
 			}
 
-			for logProcess := range tfServingProcesses {
-				process, err := createKubectlProcess(StringToLogKey(logProcess), &procAttr)
-				if err != nil {
-					socketWriterError <- err
-					return
+			logEventsOutput, err := config.AWS.CloudWatchLogsClient.FilterLogEvents(&cloudwatchlogs.FilterLogEventsInput{
+				LogGroupName:        aws.String(config.Cortex.LogGroup),
+				LogStreamNamePrefix: aws.String(prefix),
+				StartTime:           aws.Int64(lastTimestamp),
+				EndTime:             aws.Int64(time.Now().Unix() * 1000), // requires milliseconds
+				Limit:               aws.Int64(int64(maxLogLinesPerRequest)),
+			})
+
+			if err != nil {
+				if awslib.CheckErrCode(err, "ResourceNotFoundException") {
+					if !writePending {
+						writeString(socket, "pending...")
+						writePending = true
+					}
+				} else {
+					writeString(socket, "error encountered while fetching logs from cloudwatch: "+err.Error())
+					closeSocket(socket)
+					continue
 				}
-				processMap[logProcess] = process
 			}
 
-			if len(apiProcesses) != 0 && (len(tfServingProcesses) != 0 || len(initProcesses) != 0) {
-				time.Sleep(100 * time.Millisecond)
-			}
+			newEvents := strset.New()
+			for _, logEvent := range logEventsOutput.Events {
+				var log FluentdLog
+				json.Unmarshal([]byte(*logEvent.Message), &log)
 
-			for logProcess := range apiProcesses {
-				process, err := createKubectlProcess(StringToLogKey(logProcess), &procAttr)
-				if err != nil {
-					socketWriterError <- err
-					return
+				if !previousEvents.Has(*logEvent.EventId) {
+					socket.WriteMessage(websocket.TextMessage, []byte(log.Log))
+
+					if *logEvent.Timestamp > lastTimestamp {
+						lastTimestamp = *logEvent.Timestamp
+					}
 				}
-				processMap[logProcess] = process
+				newEvents.Add(*logEvent.EventId)
 			}
 
-			deleteMap := make(map[string]*os.Process, len(processesToDelete))
-
-			for podName := range processesToDelete {
-				deleteMap[podName] = processMap[podName]
-				delete(processMap, podName)
+			if len(logEventsOutput.Events) == maxLogLinesPerRequest {
+				socket.WriteMessage(websocket.TextMessage, []byte("---- Showing at most "+s.Int64(maxLogLinesPerRequest)+" lines. Visit AWS cloudwatch logs console and search for \""+prefix+"\" in log group \""+config.Cortex.LogGroup+"\" for complete logs ----"))
```

Comment:
Actually this might be an `int` rather than `int64` (my bad). Or did it convert it to int64 because of this call?